src/diffusers/models/model_loading_utils.py

```diff
@@ -136,29 +169,56 @@ def load_model_dict_into_meta(
     device: Optional[Union[str, torch.device]] = None,
     dtype: Optional[Union[str, torch.dtype]] = None,
     model_name_or_path: Optional[str] = None,
+    hf_quantizer=None,
+    keep_in_fp32_modules=None,
 ) -> List[str]:
-    device = device or torch.device("cpu")
+    device = device or torch.device("cpu") if hf_quantizer is None else device
```

Comment:
Not specific to this PR but `device = device or torch.device("cpu")` is a bit dangerous because theoretically, `0` is a valid device but it would be considered falsy. AFAICT it's not problematic for the existing code, but something to keep in mind.

Reply:
Indeed.

Reply:
I have added a comment about it too.